diff --git a/src/runtime/server/fetch.js b/src/runtime/server/fetch.js
index 1632db5b80a2e0183ed72252356153c052f1ab90..5422e8ba3dca99f01b356449121b390ffd9bba85 100644
--- a/src/runtime/server/fetch.js
+++ b/src/runtime/server/fetch.js
@@ -31,7 +31,7 @@ export function create_fetch({ event, options, state, get_cookie_header }) {
 				// Remove Origin, according to https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin#description
 				if (
 					(request.method === 'GET' || request.method === 'HEAD') &&
-					((request.mode === 'no-cors' && url.origin !== event.url.origin) ||
+					((('cf' in request ? 'cors' : request.mode) === 'no-cors' && url.origin !== event.url.origin) ||
 						url.origin === event.url.origin)
 				) {
 					request.headers.delete('origin');
@@ -48,7 +48,7 @@ export function create_fetch({ event, options, state, get_cookie_header }) {
 					// leading dot prevents mydomain.com matching domain.com
 					if (
 						`.${url.hostname}`.endsWith(`.${event.url.hostname}`) &&
-						request.credentials !== 'omit'
+						('cf' in request ? 'same-origin' : request.credentials) !== 'omit'
 					) {
 						const cookie = get_cookie_header(url, request.headers.get('cookie'));
 						if (cookie) request.headers.set('cookie', cookie);
@@ -56,7 +56,7 @@ export function create_fetch({ event, options, state, get_cookie_header }) {
 
 					let response = await fetch(request);
 
-					if (request.mode === 'no-cors') {
+					if (('cf' in request ? 'cors' : request.mode) === 'no-cors') {
 						response = new Response('', {
 							status: response.status,
 							statusText: response.statusText,
@@ -109,7 +109,7 @@ export function create_fetch({ event, options, state, get_cookie_header }) {
 					return await fetch(request);
 				}
 
-				if (request.credentials !== 'omit') {
+				if (('cf' in request ? 'same-origin' : request.origin) !== 'omit') {
 					const cookie = get_cookie_header(url, request.headers.get('cookie'));
 					if (cookie) {
 						request.headers.set('cookie', cookie);
